src/socket/set.rs

```diff
@@ -164,9 +152,9 @@ impl<'a, 'b: 'a, 'c: 'a + 'b> Set<'a, 'b, 'c> {
         Iter { lower: self.sockets.iter() }
     }
 
-    /// Iterate every socket in this set, as mutable.
+    /// Iterate every socket in this set, as SocketRef.
     pub fn iter_mut<'d>(&'d mut self) -> IterMut<'d, 'b, 'c> {
-        IterMut { lower: self.sockets.iter_mut() }
+        IterMut { lower: self.sockets.iter_mut(), index: 0 }
```

Comment:
It would be more elegant to use `.iter_mut().enumerate()` here.

Reply:
This is what I did initially, but `std::iter::Enumerate` requires `std`, so I have to do it manually.

Reply:
Apparently, I've missed `core::iter`.